File: specification/oracle/resource-manager/Oracle.Database/stable/2023-09-01/examples-js/autonomousDatabase_switchover.js
const { OracleDatabaseManagementClient } = require("@azure/arm-oracledatabase");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Perform switchover action on Autonomous Database
 *
 * @summary Perform switchover action on Autonomous Database
 * x-ms-original-file: specification/oracle/resource-manager/Oracle.Database/stable/2023-09-01/examples/autonomousDatabase_switchover.json
 */
async function performSwitchoverActionOnAutonomousDatabase() {
  const subscriptionId =
    process.env["ORACLEDATABASE_SUBSCRIPTION_ID"] || "00000000-0000-0000-0000-000000000000";
  const resourceGroupName = process.env["ORACLEDATABASE_RESOURCE_GROUP"] || "rg000";
  const autonomousdatabasename = "databasedb1";
  const body = { peerDbId: "peerDbId" };
  const credential = new DefaultAzureCredential();
  const client = new OracleDatabaseManagementClient(credential, subscriptionId);
  const result = await client.autonomousDatabases.beginSwitchoverAndWait(
    resourceGroupName,
    autonomousdatabasename,
    body,
  );
  console.log(result);
}

File: specification/oracle/resource-manager/Oracle.Database/stable/2025-03-01/examples-java/autonomousDatabase_switchover.java

import com.azure.resourcemanager.oracledatabase.models.PeerDbDetails;

/**
 * Samples for AutonomousDatabases Switchover.
 */
public final class Main {
    /*
     * x-ms-original-file: 2025-03-01/autonomousDatabase_switchover.json
     */
    /**
     * Sample code: AutonomousDatabases_Switchover.
     * 
     * @param manager Entry point to OracleDatabaseManager.
     */
    public static void
        autonomousDatabasesSwitchover(com.azure.resourcemanager.oracledatabase.OracleDatabaseManager manager) {
        manager.autonomousDatabases().switchover("rg000", "databasedb1", new PeerDbDetails().withPeerDbId("peerDbId"),
            com.azure.core.util.Context.NONE);
    }
}
